core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -380,6 +380,15 @@ class SparkContext(config: SparkConf) extends Logging {
     Utils.setLogLevel(org.apache.log4j.Level.toLevel(upperCased))
   }
 
+  def setLogLevel(logLevel: String, className: String): Unit = {
```

Comment:
I am not sure about this one. `setLogLevel` is basically just for quick debugging purpose in a shell in general. If we want some more complicated log settings, it should go for log4j.

Reply:
Yes, I just added the method to support thriftserver dynamic log level. It also looks great to support shell debugging.